crates/e2e/src/subxt_client.rs

```diff
@@ -268,6 +267,44 @@ where
             events: tx_events,
         })
     }
+
+    /// Transforms a [`ContractResult`] from a dry run into a [`Result`] type, containing
+    /// details of the [`DispatchError`] if the dry run failed.
+    fn contract_result_to_result<V>(
```

Comment:
Can't this be more generically be implemented with `From` trait?

Reply:
In this case we need the `&self` to get the metadata for decoding the `DispatchError`